### ISSUE TITLE
RKE1 Amazon EC2 clusters hang in provisioning with default node template ssh user

### ISSUE THREAD

**Issue by annablender** [rancher/dashboard#8171]
**Setup**
- Rancher version: 2.6.3, 2.7.2
- Browser type & version: Safari

**Describe the bug**

An RKE1 Amazon EC2 cluster with a node template with the default SSH user (which is `ubuntu`) hangs in provisioning trying to reach the nodes.

Default is ubuntu
<img width="1123" alt="image" src="https://user-images.githubusercontent.com/12955547/218785613-cf51e292-b1b2-45d7-9a1e-4f50ab9e01f4.png">

On 2.6.3, the Rancher AMI docs https://github.com/rancher/os/blob/master/README.md/#user-content-amazon specify the default should be `rancher.`

If you don't change the SSH user in the node template, provisioning hangs at SSH connection
<img width="1543" alt="image" src="https://user-images.githubusercontent.com/12955547/218787004-777627f1-70a6-4c03-98fe-284750c6ad2b.png">

**To Reproduce**

* Create an rke1 node template with default SSH user ubuntu
* Create rke1 cluster
* Watch `Machine Pools` tab

**Result**

RKE1 cluster hangs trying to connect to the nodes via SSH

**Expected Result**

RKE1 cluster should provision successfully.

**Comment by gaktive** [comment]
I think a fix is only viable for 2.7.x since 2.6.11 is set to ship before the end of the month and QA needs time, then 2.6.x is scheduled for security updates only.

@Anna-Blendermann can you confirm whether 2.6.3 is accurate and if this is the case with 2.6.10? I'm curious if this is regression that's been around for a long time.

**Comment by annablender** [comment]
Yes, I was reproducing a different issue on 2.6.3 this morning and saw this. I'll try on 2.6.10

**Comment by annablender** [comment]
Issue confirmed on 2.6.10

Repro steps
* Run rancher instance 2.6.10
* Create aws cloud creds
* Create 2 rke1 node templates, 1 with the default SSH user `ubuntu` and one with SSH user `rancher`
* Watch the `Machine pools` tab for both clusters and see that the cluster with the SSH user `rancher` is the only one that provisions successfully. The other cluster hangs at SSH

<img width="1541" alt="image" src="https://user-images.githubusercontent.com/12955547/218819143-2169008e-c4c0-47f1-af7b-de43e0f11d40.png">

**Comment by gaktive** [comment]
Thanks @Anna-Blendermann 
For UI, if this is quick for 2.7.x, then let's get a backport for 2.6.11 pronto since the deadline for that version is coming in fast

**Comment by nwmac** [comment]
This is a backend issue - the schema for `amazonec2config` contains the default for the `sshUser` field:

<img width="1074" alt="image" src="https://user-images.githubusercontent.com/1955897/219118166-7f6fb57f-c444-49ca-b92b-26061bcfad36.png">

@gaktive We should transfer this issue to `rancher/rancher`
